barcode/main.py

```diff
@@ -94,16 +96,15 @@ def get_user_input(input_value):
     Question user for input
     :return:
     """
-    if input_value:
-        if check_input_validity(input_value):
-            return input_value
-    while True:
+    pred = lambda iv: isinstance(iv, type(None)) or not check_input_validity(iv)
+    
+    while pred(input_value):
         try:
-            current_code = raw_input("Your code (max len 13 char): ")
+            input_value = raw_input("Your code (max length 13 char from [0, 9]): ")
```

Comment:
Python 2 is dead, I don't think that we want to keep this compatibility more longer. Maybe it's the right time to clean it.

Reply:
> 
> 
> Thank you for dusting off this project :)

No worries, just hope they were/will be good contributions!